src/main/kotlin/dsl/domain/Skills.kt

```diff
@@ -0,0 +1,23 @@
+package dsl.domain
+
+/**
+ * Created by Jaesungchi on 2022.06.03..
+ */
+class Skills(
+    softs: List<Skill> = emptyList(),
+    hards: List<Skill> = emptyList()
+) {
+    private val _softs = softs.toMutableList()
+    val softs: List<Skill> get() = _softs.toList()
+
+    private val _hards = hards.toMutableList()
+    val hards: List<Skill> get() = _hards.toList()
+
+    fun addSoft(skill: Skill) {
+        _softs.add(skill)
+    }
+
+    fun addHard(skill: Skill) {
+        _hards.add(skill)
+    }
+}
```

Comment:
아래 코드로 개선해 볼 수 있을까요?
```suggestion
data class Skills(private val skill: MutableList<Skill> = mutableListOf()) : MutableList<Skill> by skill {

    fun soft(name: String) {
        skill.add(Skill.Soft(name))
    }

    fun hard(name: String) {
        skill.add(Skill.Hard(name))
    }
}
```